subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/SnapshotTaskInputsOperationIntegrationTest.groovy

```diff
@@ -272,28 +272,28 @@ class SnapshotTaskInputsOperationIntegrationTest extends AbstractIntegrationSpec
                 path == file("a/src/main/java").absolutePath
                 children.size() == 3
                 with(children[0]) {
-                    path == "A.java"
-                    hash != null
-                }
-                with(children[1]) {
-                    path == "B.java"
-                    hash != null
-                }
-                with(children[2]) {
                     path == "a"
                     children.size() == 2
                     with(children[0]) {
-                        path == "A.java"
-                        hash != null
-                    }
-                    with(children[1]) {
                         path == "a"
                         children.size() == 1
                         with(children[0]) {
                             path == "A.java"
                             hash != null
                         }
                     }
+                    with(children[1]) {
+                        path == "A.java"
+                        hash != null
+                    }
+                }
+                with(children[1]) {
+                    path == "A.java"
+                    hash != null
+                }
+                with(children[2]) {
+                    path == "B.java"
+                    hash != null
```

Comment:
❓This changed because the ordering of the file walking has changed, right?

Reply:
Exactly. `A.java` is not before `a` any more, but the other way round.